interview-questions.md

```diff
@@ -8,36 +8,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: You can add the foreign key by using migration and then creating a new column in the students table and it will be on the student model. The name of the foreign key would be student_id.
 
-Researched answer:
+Researched answer: The file migration command will fix the mistake. Even after a database is created your able to generate the rails command to create a migration which allows additional columns to be created. When generating the migrate command it'll be displayed in snake_case, the command to achieve this is: 'rails generate migration' add_columns_to_list. You can outline the specific changes/additions you want to make to the database with 'add_column :lists, :item, :string' command' With the _ syntax upon the creation of the student model the foreign key would be student_id.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The 'show', 'edit' and 'destroy' route without an id there's no way of knowing which is being performed.
+
+Researched answer: They need a unique identifier to show specific content, edit needs a unique identifier to outline which changes need to be made to a specific identifier. The destroy route is like the edit route, it needs a unique identifier to outline what needs to be deleted. Parameters are required for because with these specific routes their inputs have different outcomes that affect the application, therfore parameters are used to affect the outcome of the routes. 
 
-Researched answer:
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Rails new, creates a new application. The db migrate allow edits like addition to add columns to a current database. The db generate, is used to create a new database.
```

Comment:
Good job.  Just a note:  rails db:create adds a database.  I encourage your to look up `rails generate` and see what those commands are.